pkg/api/validate/api.go

```diff
@@ -51,6 +51,9 @@ func (v *APIValidator) validateUpdateContainerService(cs, oldCs *api.OpenShiftMa
 	// validating ProvisioningState is the RP's responsibility
 	old.Properties.ProvisioningState = cs.Properties.ProvisioningState
 
+	old.Properties.MonitorProfile.WorkspaceID = cs.Properties.MonitorProfile.WorkspaceID
+	old.Properties.MonitorProfile.WorkspaceKey = cs.Properties.MonitorProfile.WorkspaceKey
```

Comment:
MonitorProfile is updatable, so you can enable/disable Monitoring by doing an update.